mmdet/core/hook/wandblogger_hook.py

```diff
@@ -0,0 +1,455 @@
+        self.data_table = self.wandb.Table(columns=columns)
+
+    def _init_pred_table(self):
+        """Initialize the W&B Tables for model evaluation."""
+        columns = ['epoch', 'image_name', 'ground_truth', 'prediction'] + list(
+            self.class_id_to_label.values())
+        self.eval_table = self.wandb.Table(columns=columns)
+
+    def _add_ground_truth(self):
+        # Get image loading pipeline
+        from mmdet.datasets.pipelines import LoadImageFromFile
+        transforms = self.val_dataset.pipeline.transforms
+        for transform in transforms:
+            if isinstance(transform, LoadImageFromFile):
+                img_loader = transform
+        if 'img_loader' not in locals():
+            warnings.warn(
+                'LoadImageFromFile is required to add images '
+                'to W&B Tables.', UserWarning)
+            self.log_evaluation = False
+
+        # Determine the number of samples to be logged.
+        num_total_images = len(self.val_dataset)
+        if self.num_eval_images > num_total_images:
+            warnings.warn(
+                'The num_eval_images is greater than the total number '
+                'of validation samples. The complete validation set '
+                'will be logged.', UserWarning)
+        self.num_eval_images = min(self.num_eval_images, num_total_images)
+
+        classes = self.val_dataset.get_classes()
+        self.class_id_to_label = {id: name for id, name in enumerate(classes)}
+        self.class_set = self.wandb.Classes([{
+            'id': id,
+            'name': name
+        } for id, name in self.class_id_to_label.items()])
+
+        img_prefix = self.val_dataset.img_prefix
+
+        for idx in range(self.num_eval_images):
+            img_info = self.val_dataset.data_infos[idx]
+            img_meta = img_loader(
+                dict(img_info=img_info, img_prefix=img_prefix))
+
+            # Get image and convert from BGR to RGB
+            image = img_meta['img'][..., ::-1]
+            image_name = img_info['filename']
+
+            data_ann = self.val_dataset.get_ann_info(idx)
+            bboxes = data_ann['bboxes']
+            labels = data_ann['labels']
+
+            assert len(bboxes) == len(labels)
+            box_data = self._get_wandb_bboxes(bboxes, labels, mode='gt')
+
+            boxes = {
+                'ground_truth': {
+                    'box_data': box_data,
+                    'class_labels': self.class_id_to_label
+                }
+            }
+
+            self.data_table.add_data(
+                image_name,
+                self.wandb.Image(image, boxes=boxes, classes=self.class_set))
+
+    def _log_predictions(self, results, epoch):
+        table_idxs = self.data_table_ref.get_index()
+        assert len(table_idxs) == self.num_eval_images
+
+        for ndx in table_idxs:
+            result = results[ndx]
+            assert len(result) == len(self.class_id_to_label)
```

Comment:
mmdetection also support instance segmentation model like mask rcnn. Their outputs is an tuple which is an combination of bbox results and segm results. Could you also support instance segmentation outputs in this pr?

Reply:
I can work on supporting instance segmentation as well. But can you point me to an example/tutorial that used MMDetection for the segmentation task?

PS: W&B lacks good support for "instance" segmentation at the moment but I can try making a workaround.

Reply:
Also does MMDetection support "semantic" segmentation?

Reply:
You can refer to the [show_result](https://github.com/open-mmlab/mmdetection/blob/master/mmdet/models/detectors/base.py#L271) function for the format of mask results. If you have any questions, you can directly point them out here.

Reply:
Semantic segmentation is not supported.

Reply:
As mentioned above W&B lacks first-class support for "instance" segmentation and for this reason I will be adding this feature once W&B build support for this. 

For now, the `results` format is handled properly. Thanks for pointing this out.